daemon/container.go

```diff
@@ -1627,3 +1618,53 @@ func (container *Container) monitorExec(execConfig *execConfig, callback execdri
 
 	return err
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	paths, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(paths) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := findSignificantParent(paths)
+
+	relative := make([]string, len(paths))
+
+	p := fmt.Sprintf("%s%s", baseDir, string(filepath.Separator))
+	for i, f := range paths {
+		r := f[len(p):]
+
+		if r != "" {
+			relative[i] = r
+		}
+	}
+
+	return &archiveFilter{baseDir, relative}, nil
+}
+
+func findSignificantParent(paths []string) string {
+	baseDir := filepath.Dir(paths[0])
+
+	for i := 1; i < len(paths); i++ {
+		p := paths[i]
+		if strings.HasPrefix(baseDir, p) {
+			baseDir = p
+		} else if !strings.HasPrefix(p, baseDir) {
+			d1 := filepath.Dir(baseDir)
```

Comment:
Would you need to keep moving up the tree looking for common paths instead of just one level?  I would think that if baseDir was "/a/b/c" and p was "/d/e/f" then baseDir after this if-statement would need to be just "/".  Right?